src/server/routes/__tests__/catchments.unit.spec.js

```diff
@@ -0,0 +1,99 @@
+import { Catchment } from '../../../entities/index.js'
+import logger from '../../../utils/logger-utils.js'
+import routes from '../catchments.js'
+
+jest.mock('../../../entities/index.js')
+jest.mock('../../../utils/logger-utils.js')
+
+describe('catchments.unit', () => {
+  describe('GET /catchments', () => {
+    const h = {
+      response: jest.fn().mockReturnThis(),
+      code: jest.fn()
+    }
+
+    const catchmentData = [
+      {
+        id: '2',
+        name: 'Adur',
+        createdAt: '2018-11-07T10:00:00.000Z',
+        updatedAt: '2018-11-07T10:00:00.000Z'
+      },
+      {
+        id: '6',
+        name: 'Annas',
+        createdAt: '2018-11-07T10:00:00.000Z',
+        updatedAt: '2018-11-07T10:00:00.000Z'
+      }
+    ]
+
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it('should return a 200 status code if the call to fetch all catchments is successful', async () => {
+      Catchment.findAll.mockResolvedValueOnce(catchmentData)
+
+      const handler = routes[0].options.handler
+      await handler({}, h)
+
+      expect(h.code).toHaveBeenCalledWith(200)
+    })
+
+    it('should return all catchments if the call to fetch all catchments is successfull', async () => {
+      Catchment.findAll.mockResolvedValueOnce(catchmentData)
+
+      const handler = routes[0].options.handler
+      await handler({}, h)
+
+      expect(h.response).toHaveBeenCalledWith(
+        expect.objectContaining({
+          _embedded: {
+            catchments: catchmentData
+          }
+        })
+      )
+    })
+
+    it('should log an error if an error occurs while fetching catchments', async () => {
+      const error = new Error('Database error')
+      Catchment.findAll.mockRejectedValueOnce(error)
+
+      const handler = routes[0].options.handler
+      await handler({}, h)
+
+      expect(logger.error).toHaveBeenCalledWith(
+        'Error fetching catchments:',
+        error
+      )
+    })
+
+    it('should return 500 if an error occurs while fetching catchments', async () => {
+      const error = new Error('Database error')
+      Catchment.findAll.mockRejectedValueOnce(error)
+
+      const handler = routes[0].options.handler
+      await handler({}, h)
+
+      expect(h.response).toHaveBeenCalledWith(
+        expect.objectContaining({
+          error: 'Unable to fetch catchments'
+        })
+      )
+    })
+
+    it('should return the error response in the body if an error occurs while fetching catchments', async () => {
+      const error = new Error('Database error')
+      Catchment.findAll.mockRejectedValueOnce(error)
+
+      const handler = routes[0].options.handler
+      await handler({}, h)
+
+      expect(h.response).toHaveBeenCalledWith(
+        expect.objectContaining({
+          error: 'Unable to fetch catchments'
+        })
+      )
+    })
+  })
+})
```

Comment:
Sorry to do this to you, but in my first review I missed something in catchments.unit.spec.js - you’re setting up h as the response toolkit, but then re-use it in all your tests. We have a rule now of ‘fresh test, fresh mocks’ - it’s easily fixed by doing:
```
const getResponseToolkit = () => ({
  response: jest.fn().mockReturnThis(),
  code: jest.fn()
})
```
Rather than getting a reference to handler in every test, it’d be ok to get one reference after your imports and re-use it. Finally, you're doing the same thing with catchmentData - so simply turn it into a function so there's a fresh copy of the mock data for each test.